ansible_collections/studyly/git_mactp/plugins/modules/git_mactp.py

```diff
@@ -277,17 +287,27 @@ def main():
             )
 
     git = Git(module)
-    changed_files = git.status()
+    # checkout branch if parameter was given
+    if branch:
+        git.checkout()
 
-    if changed_files:
-        if pull:
-            result.update(git.pull())
+    if pull:
+        result.update(git.pull())
+    
+    changed_files = git.status()
 
+    if merge:
+        result.update(git.merge())
+    elif changed_files and add:
         git.add()
         result.update(git.commit())
+    elif tag:
+        result.update(git.tag())
+    else:
+        push = False
 
-        if push:
-            result.update(git.push())
+    if push:
+        result.update(git.push())
         result["changed"] = True
```

Comment:
I know we talked and decided on this being fine, but as I saw the actual code of the "push" function, why is it necessary to overwrite `changed` here? The function itself, if successful, will always return `changed == true` in its current shape. If refactored as suggested there, it might not. But it still always should return the proper value. Maybe I'm just overlooking something here.